File: Find-the-Highest-Altitude.java
1class Solution {
2    public int largestAltitude(int[] gain) {
3        int res = gain[0];
4        int curr = gain[0];
5        for (int i = 1; i < gain.length; i++){
6            int nxt = curr + gain[i];
7            res = Math.max(res, nxt);
8            curr = nxt;
9        }
10
11        if (res < 0)
12        return 0;
13
14        return res;
15    }
16}

File: DCP-06-26/Find-the-Highest-Altitude.py
1class Solution:
2    def largestAltitude(self, gain: List[int]) -> int:
3        res = 0
4        temp = 0
5        for num in gain:
6            curr = temp + num
7            temp = curr
8            if curr > res:
9                res = curr
10            
11        return res